### ISSUE TITLE
Add a builtin to test the existence of a field

### ISSUE THREAD

**Issue by cueckoo** [cue-lang/cue#140]
_Originally opened by @nlewo in https://github.com/cuelang/cue/issues/140_

It is currently possible to manipulate optional fields with field comprehension but it's a bit hacky:/

For instance:
```
a : {opt? : string, required : "required"}
b : { for k, v in a if k == "opt" { optional : a.opt }} & a
```
Evaluates to:
```
b: {
    required: "required"
}
```

And 
```
a : {opt? : string, required : "required"}
a : { opt : "my-string" }
b : { for k, v in a if k == "opt" { optional : a.opt }} & a
```
Evaluates to 
```
b: {
    required: "required"
    optional:      "my-string"
}
```

So, this works well, but it would be much more intuitive with a `hasField`builtin which would allow us to define `b` like:
```
b : if hasField(a, "opt") { optional : a.opt } & a
```

**Comment by cueckoo** [comment]
_Original reply by @mpvl in https://github.com/cuelang/cue/issues/140#issuecomment-538416872_

Did you try:
```
if a["opt"] != _|_ {
	optional : a.opt
}
```

**Comment by cueckoo** [comment]
_Original reply by @mpvl in https://github.com/cuelang/cue/issues/140#issuecomment-538455520_

It seems that for now `_|_` seems sufficient. A query language may give additional solutions. It seems to make sense to hold off until then.

Not applicable here, but also note that defaults can be used for coalescing: *a["opt"] | "default".

**Comment by cueckoo** [comment]
_Original reply by @nlewo in https://github.com/cuelang/cue/issues/140#issuecomment-539390842_

No, i didn't try your suggestion which works well!
But, I think I had tried:
```
if a.opt != _|_ ...
```
which doesn't work.

In another case I don't have a top-level attribute and I have been able to do:
```
if "\(opt)" != _|_ ....
```

Maybe a section `tips and tricks` would be useful for this kind of stuffs.

Thx for you help!
